runhouse/servers/cluster_servlet.py

```diff
@@ -214,105 +214,112 @@ async def aclear_all_references_to_env_servlet_name(self, env_servlet_name: str)
+
+        resource_info = dict(status)
+        env_servlet_processes = dict(resource_info.pop("env_servlet_processes"))
+        status_data = {
+            "status": ResourceServerStatus.running,
+            "resource_type": status.cluster_config.get("resource_type"),
+            "resource_info": resource_info,
+            "env_servlet_processes": env_servlet_processes,
+        }
+        client = httpx.AsyncClient()
+        return await client.post(
+            f"{api_server_url}/resource/{cluster_uri}/cluster/status",
+            data=json.dumps(status_data),
+            headers=rns_client.request_headers(),
+        )
 
     async def aperiodic_cluster_checks(self):
+        """Periodically check the status of the cluster, gather metrics about the cluster's utilization & memory,
+        and save it to Den."""
         # Delay the start of post_status_thread, so we'll finish the cluster startup properly
         await asyncio.sleep(SCHEDULERS_DELAY)
+        cluster_config = await self.aget_cluster_config()
+        interval_size = cluster_config.get(
+            "status_check_interval", DEFAULT_STATUS_CHECK_INTERVAL
+        )
         while True:
             try:
-
-                cluster_config = await self.aget_cluster_config()
-                interval_size = cluster_config.get(
-                    "status_check_interval", DEFAULT_STATUS_CHECK_INTERVAL
-                )
-
                 # Only if one of these is true, do we actually need to get the status from each EnvServlet
-                should_send_status_and_logs_to_den = (
+                should_send_status_and_logs_to_den: bool = (
                     configs.token is not None and interval_size != -1
                 )
-                should_update_autostop = self.autostop_helper is not None
-                if should_send_status_and_logs_to_den or should_update_autostop:
+                should_update_autostop: bool = self.autostop_helper is not None
+
+                if (
+                    not should_send_status_and_logs_to_den
+                    and not should_update_autostop
+                ):
+                    break
+
+                logger.debug("Performing cluster checks")
+                status: ResourceStatusData = await self.astatus()
+                if should_update_autostop:
+                    logger.debug("Updating autostop")
+                    await self._update_autostop(status)
+
+                if not should_send_status_and_logs_to_den:
+                    break
+
+                logger.debug("Sending cluster status to Den")
+                cluster_rns_address = cluster_config.get("name")
+                cluster_uri = rns_client.format_rns_address(cluster_rns_address)
+                api_server_url = status.cluster_config.get(
+                    "api_server_url", rns_client.api_server_url
+                )
+
+                resp = await ClusterServlet.save_status_metrics_to_den(
```

Comment:
Small nit: maybe rename `resp` to `sent_status`? so it would match `send_logs` (line 299)?

Reply:
i changed it tho to be a response object, instead of a status code, so that we can surface the response json in the error

Reply:
ok cool, in that case my next comment is irrelevant, can leave `status_code` as is :)